include/openPMD/RecordComponent.hpp

```diff
@@ -29,10 +29,41 @@
 #include <queue>
 #include <string>
 #include <stdexcept>
+#include <vector>
+#include <array>
 
 
 namespace openPMD
 {
+//! @{
+/** Share ownership with a raw pointer
+ *
+ * Helper function to share write ownership
+ * unprotected and without reference counting with a
+ * raw pointer or stdlib container.
+ */
+template< typename T >
+std::shared_ptr< T >
+shareRaw( T* x )
```

Comment:
possible naming: `storeRaw`

since we will need another wrapper for loads

Reply:
Adapting from future C++ lingo, we might want to call this a [```span```](http://en.cppreference.com/w/cpp/container/span) (I'd still prefer the name [```array_view```](http://open-std.org/Jtc1/sc22/wg21/docs/papers/2017/p0122r4.pdf), but well :man_shrugging:).
One could think about ```shared_span```/```unique_span``` and ```readonly_span```/```readwrite_span```

Reply:
Sidenote - ```span``` is available in [GSL](https://github.com/Microsoft/GSL/blob/master/include/gsl/span) should we want to use it.

Reply:
Fits quite good as well, similar to the `ContiguousIterator` concept discussed earlier.